### PR TITLE
[develop] Fixing several issues, including 966 (bash octal issue); add new winter weather verification test with staged data

### LINKED ISSUE #966: Incorrect octal notation causing ensemble vx to fail
## Expected behavior
When running verification on ensembles with membership >=8, verification should successfully complete.

## Current behavior
The current way for calculating `ensmem_indx=$(printf "%0${VX_NDIGITS_ENSMEM_NAMES}d" "${ENSMEM_INDX}")` is causing octal issues for members 8 and 9.

## Machines affected
Any machine running ensemble verification will be affected.

## Steps To Reproduce
Run WE2E test: `ufs-srweather-app/tests/WE2E/test_configs/verification/config.MET_ensemble_verification.yaml`
But change `NUM_ENS_MEMBERS` from 2 to 10.

## Detailed Description of Fix (optional)
Perhaps there are better ways to address this, but this is one option of addressing the issue: `ensmem_indx=$(printf "%0${VX_NDIGITS_ENSMEM_NAMES}d" $(( 10#${ENSMEM_INDX})))`

This will need to be updated in `exregional_check_post_output.sh`, `exregional_run_met_pcpcombine.sh`, and `exregional_run_met_gridstat_or_pointstat_vx.sh`.

Thanks to @mkavulich for helping with this!

### LINKED ISSUE #978: Should "EXPT_SUBDIR" be a mandatory variable?
## Description

Currently, [the Users Guide](https://ufs-srweather-app.readthedocs.io/en/develop/CustomizingTheWorkflow/ConfigWorkflow.html#directory-parameters) (as well as in-line documentation in `config_defaults.yaml`) states that `EXPT_SUBDIR` is a **mandatory** variable: it must be specified by the user. This seems draconian to me, as the only thing it is used for is setting the subdirectory where an experiment is run. And what's more, if the user specifies their own custom full-path value for `EXPTDIR`, it doesn't get used at all!

As an additional headache, there is currently a bug in the workflow generation that fails in a very messy and unintuitive way if `EXPT_SUBDIR` is not set:

```
ln: target ‘}}/launch_FV3LAM_wflow.sh’ is not a directory
  File "/scratch1/BMC/hmtb/kavulich/UFS/workdir/issue_966/ufs-srweather-app/ush/./generate_FV3LAM_wflow.py", line 768, in <module>
    expt_dir = generate_FV3LAM_wflow(USHdir, wflow_logfile, pargs.debug)
  File "/scratch1/BMC/hmtb/kavulich/UFS/workdir/issue_966/ufs-srweather-app/ush/./generate_FV3LAM_wflow.py", line 143, in generate_FV3LAM_wflow
    create_symlink_to_file(
  File "/scratch1/BMC/hmtb/kavulich/UFS/workdir/issue_966/ufs-srweather-app/ush/python_utils/create_symlink_to_file.py", line 53, in create_symlink_to_file
    ln_vrfy(f"-sf {relative_flag} {target} {symlink}")
  File "/scratch1/BMC/hmtb/kavulich/UFS/workdir/issue_966/ufs-srweather-app/ush/python_utils/filesys_cmds_vrfy.py", line 41, in ln_vrfy
    return cmd_vrfy("ln", *args)
  File "/scratch1/BMC/hmtb/kavulich/UFS/workdir/issue_966/ufs-srweather-app/ush/python_utils/filesys_cmds_vrfy.py", line 20, in cmd_vrfy
    print_err_msg_exit(f"System call '{cmd}' failed.")
  File "/scratch1/BMC/hmtb/kavulich/UFS/workdir/issue_966/ufs-srweather-app/ush/python_utils/print_msg.py", line 20, in print_err_msg_exit
    traceback.print_stack(file=sys.stderr)
FATAL ERROR: System call 'ln -sf  /scratch1/BMC/hmtb/kavulich/UFS/workdir/issue_966/ufs-srweather-app/ush/launch_FV3LAM_wflow.sh /scratch1/BMC/hmtb/kavulich/UFS/workdir/issue_966/expt_dirs/{{ EXPT_SUBDIR }}/launch_FV3LAM_wflow.sh' failed.
Exiting with nonzero status.

*********************************************************************
FATAL ERROR:
Experiment generation failed. See the error message(s) printed below.
For more detailed information, check the log file from the workflow
generation script: /scratch1/BMC/hmtb/kavulich/UFS/workdir/issue_966/ufs-srweather-app/ush/log.generate_FV3LAM_wflow
*********************************************************************

Traceback (most recent call last):
  File "/scratch1/BMC/hmtb/kavulich/UFS/workdir/issue_966/ufs-srweather-app/ush/./generate_FV3LAM_wflow.py", line 768, in <module>
    expt_dir = generate_FV3LAM_wflow(USHdir, wflow_logfile, pargs.debug)
  File "/scratch1/BMC/hmtb/kavulich/UFS/workdir/issue_966/ufs-srweather-app/ush/./generate_FV3LAM_wflow.py", line 143, in generate_FV3LAM_wflow
    create_symlink_to_file(
  File "/scratch1/BMC/hmtb/kavulich/UFS/workdir/issue_966/ufs-srweather-app/ush/python_utils/create_symlink_to_file.py", line 53, in create_symlink_to_file
    ln_vrfy(f"-sf {relative_flag} {target} {symlink}")
  File "/scratch1/BMC/hmtb/kavulich/UFS/workdir/issue_966/ufs-srweather-app/ush/python_utils/filesys_cmds_vrfy.py", line 41, in ln_vrfy
    return cmd_vrfy("ln", *args)
  File "/scratch1/BMC/hmtb/kavulich/UFS/workdir/issue_966/ufs-srweather-app/ush/python_utils/filesys_cmds_vrfy.py", line 20, in cmd_vrfy
    print_err_msg_exit(f"System call '{cmd}' failed.")
  File "/scratch1/BMC/hmtb/kavulich/UFS/workdir/issue_966/ufs-srweather-app/ush/python_utils/print_msg.py", line 24, in print_err_msg_exit
    sys.exit(1)
SystemExit: 1
```

## Solution
One of two things should be changed here:

1. `EXPT_SUBDIR` should get a default value. The literal string "experiment" seems appropriate, but it could really be anything so long as it's documented appropriately.
2. `generate_FV3LAM_wflow()` should fail with a verbose/explicit error message if `EXPT_SUBDIR` is not specified. This seems as if it was intended behavior, but was not implemented correctly: there is a check for empty `EXPT_SUBDIR` in `setup.py`, but it is not empty if not specified in config.yaml, but rather gets set to the literal string `{{ EXPT_SUBDIR }}`. 

I prefer the first solution, but either would be acceptable.

## Requirements**
Users should not receive strange errors if they do not set set EXPT_SUBDIR.

## Acceptance Criteria (Definition of Done)
Implement either one of the suggested solutions above

### PR DESCRIPTION
**Pull request by mkavulich**
## DESCRIPTION OF CHANGES: 
This started as a fairly simple change (don't they always) to fix Issue #966 and add a new winter weather verification test in support of the RRFS agile framework project and has since snowballed into solving a number of outstanding and newly discovered issues.

### New test
 - The new test `MET_ensemble_verification_winter_wx` is added. This test will exercise a number of yet-untested capabilities in the workflow, including a 10-member ensemble, snowfall verification with staged data (so can be run on all platforms, not just Jet and Hera), and several SPP settings.
 - As part of this new test, snowfall observations will now be staged on all tier-1 platforms, as well as netCDF GFS data and other observation types, all for the date 2022020300

### Resolved issues
 - #966 Resolved: In several locations, an explicit conversion is done to `ENSMEM_INDX` to ensure it is a base-10 integer, to avoid problems with bash interpreting numbers with a leading zero as octal.
 - #978 resolved: per discussion in a recent SRW code management meeting, give `EXPT_SUBDIR` a default value "experiment" to avoid unnecessary complications and work for users. Additionally, the default behavior if an experiment directory already exists is changed to "quit" rather than "delete"
 - Issue mentioned in [this discussion](https://github.com/orgs/ufs-community/discussions/33); the setting `fhzero=6` is removed from the weather model namelist for CCPP suite `FV3_GFS_v17_p8`, which allows precipitation and other accumulations to be made every hour rather than 6 hours (SRW output is always hourly, so this makes sense). Also, update `diag_table.FV3_GFS_v17_p8` so that all output files will be hourly
 - Per discussion in #939, remove an unnecessary special case in `get_crontab_contents.py` for Derecho

### Other fixes
 - Some old files for unsupported/removed CCPP suites are removed
 - Add some missing task dependencies for retrieving verification obs

### General improvements
 - Many improvements to verification obs-pulling task
   - NDAS observations are now retrieved for forecast hour zero, and a better obs file is retrieved for major obs times (00z, 06z, 12z, 18z) per EMC guidance
   - Better in-line comments/documentation
   - Standardize order and messaging for file-on-disk checks across all observation types
 - Added explanatory comments for reflectivity field in diag_table files
 - Update `diag_table.FV3_GFS_v17_p8` so that all output files will be hourly
 - Simplify task dependencies that rely on staged verification observations; these "get_obs" tasks should always be run (they check that the data exists before trying to retrieve it), so no need to make the dependency conditional
 - Add a check in `monitor_jobs.py` to ensure the yaml file does not contain duplicate experiment directories
 - Make sure the key in the experiment dictionary used by is unique by appending the current date/time to the exptdir name; additionally, set this key as the `WORKFLOW_ID` variable (so that it could be used in the workflow if necessary).

### Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## TESTS CONDUCTED: 
Ran fundamental tests on several platforms. Ran some manual tests for the above-mentioned WE2E and monitor_jobs features, including catting together several WE2E yaml files in a way that previously failed to ensure they now succeed.

Ran all verification tests (including new test) on Hera, Orion, and Derecho to ensure nothing changed unexpectedly. I also compared existing tests to develop on Hera, and confirmed that the additional NDAS obs are now being pulled and used for verification tasks, which resulted in expected differences in verification results.

- [x] hera.intel
  - Fundamental tests
  - All verification tests
- [x] orion.intel
  - Fundamental tests
  - All verification tests (except `MET_verification_winter_wx` and `MET_ensemble_verification_only_vx_time_lag`, which only work on Jet and Hera)
- [ ] hercules.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [x] derecho.intel
  - Fundamental tests
  - All verification tests (except `MET_verification_winter_wx` and `MET_ensemble_verification_only_vx_time_lag`, which only work on Jet and Hera)
- [ ] gaea.intel
- [ ] gaeac5.intel
- [ ] jet.intel
- [ ] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [x] fundamental test suite
- [ ] comprehensive tests

## DEPENDENCIES:
None

## DOCUMENTATION:
Working on updates to documentation (mostly for the new test); will merge when ready

## ISSUE: 
 - Resolves #966 
 - Resolves #978 
 - Resolves issue from discussion: https://github.com/orgs/ufs-community/discussions/33

## CHECKLIST
- [x] My code follows the style guidelines in the Contributor's Guide
- [x] I have performed a self-review of my own code using the Code Reviewer's Guide
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

## CONTRIBUTORS:
Thanks to @EdwardSnyder-NOAA for help staging the new test data

### CODE REVIEW

File: parm/diag_table.FV3_GFS_v17_p8 [hunk 1]
```diff
@@ -154,6 +154,8 @@
 "gfs_phys",  "BC_AOD_550",    "bc_aod550",     "fv3_history2d",      "all",  .false.,  "none",  2
 "gfs_phys",  "OC_AOD_550",    "oc_aod550",     "fv3_history2d",      "all",  .false.,  "none",  2
 "gfs_phys",  "SS_AOD_550",    "ss_aod550",     "fv3_history2d",      "all",  .false.,  "none",  2
+# Reflectivity from Thompson microphysics
```
Comment (JeffBeck-NOAA) [inline parm/diag_table.FV3_GFS_v17_p8:157]: Shouldn't this be 'GFDL microphysics'?  This is a GFS SDF.

File: tests/WE2E/test_configs/custom_grids/config.custom_ESGgrid_Great_Lakes_snow_8km.yaml [hunk 2]
```diff
@@ -2,8 +2,8 @@ metadata:
   description: |-
     This test checks the capability of the workflow to run verification tasks
     on a user-specified grid. This also serves as a winter snowstorm case for
-    checking snowfall verification statistics, and a test for custom domains
-    with staged netCDF data.
+    checking snowfall verification statistics using observations retrieved from
+    HPSS, and a test for custom domains with RAP data retrieved from HPSS
 user:
```
Comment (JeffBeck-NOAA) [inline tests/WE2E/test_configs/custom_grids/config.custom_ESGgrid_Great_Lakes_snow_8km.yaml:7]: Could HRRR ICs be used for this domain too?  Just curious why RAP is mentioned by itself.

File: ush/get_crontab_contents.py [hunk 3]
```diff
@@ -36,7 +36,7 @@ def get_crontab_contents(called_from_cron, machine, debug):
     # call the system version of crontab at /usr/bin/crontab.
     #
     crontab_cmd = "crontab"
-    if machine == "CHEYENNE" or machine == "DERECHO":
+    if machine == "CHEYENNE":
         if called_from_cron:
```
Comment (JeffBeck-NOAA) [inline ush/get_crontab_contents.py:40]: Is there no Derecho support for the App at the moment?

**Review by JeffBeck-NOAA — APPROVED** [review]

File: parm/diag_table.FV3_GFS_v17_p8 (hunk 1 above)
Comment (michelleharrold) [reply parm/diag_table.FV3_GFS_v17_p8:157]: GFSv17 P8 is testing with Thompson!

File: ush/get_crontab_contents.py (hunk 3 above)
Comment (MichaelLueken) [reply ush/get_crontab_contents.py:40]: The App currently runs on Derecho.  Given the decommissioning of Cheyenne, I think that `DERECHO` should be kept, while `CHEYENNE` should be removed here.

Comment (mkavulich) [reply ush/get_crontab_contents.py:40]: Derecho does not suffer from the same issue that this logic accounts for, see my PR description for why this fix was included (and on a related note, #998).  Cheyenne functionality should be removed in its entirety in a separate PR, I feel that's beyond the scope of this PR.

File: tests/WE2E/test_configs/custom_grids/config.custom_ESGgrid_Great_Lakes_snow_8km.yaml (hunk 2 above)
Comment (mkavulich) [reply tests/WE2E/test_configs/custom_grids/config.custom_ESGgrid_Great_Lakes_snow_8km.yaml:7]: I was simply making the description more verbose: this test specifically retrieves RAP data, but GFS and HRRR (and GEFS/GDAS for that matter) can also be retrieved.

Now that you mention it though I'm pretty sure this custom domain might extend a bit north of the HRRR domain, so those ICs might not be possible for this specific domain.

File: parm/diag_table.FV3_GFS_v17_p8 (hunk 1 above)
Comment (mkavulich) [reply parm/diag_table.FV3_GFS_v17_p8:157]: That's correct @michelleharrold. I'm not sure if it was different previously, but from what I can see currently the `FV3_GFS_v17_p8` suite uses Thompson microphysics: https://github.com/NOAA-EMC/fv3atm/blob/bba399053d3939241938f19ee598895eea54fd65/ccpp/suites/suite_FV3_GFS_v17_p8.xml#L76

**Review by MichaelLueken — APPROVED** [review]
@mkavulich -

Thanks for removing the last vestiges of the `FV3_GFS_2017_gfdlmp`/`_regional` CCPP suites from the repository!  I was able to successfully test that the change in `ush/get_crontab_contents.py` works while using `run_WE2E_tests.py` on Derecho and using the `--launch cron` option.  I was also able to successfully run the new `MET_ensemble_verification_winter_wx` test on Gaea C5:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
MET_ensemble_verification_winter_wx_20240104094734                 COMPLETE             208.52
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             208.52
```
Approving this PR now.

**Comment by MichaelLueken** [comment]
The coverage WE2E tests were manually ran on Derecho and they all successfully passed:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
custom_ESGgrid_IndianOcean_6km_20240104115217                      COMPLETE              24.66
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot_20  COMPLETE              45.55
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16_2024010411521  COMPLETE              45.90
grid_RRFS_CONUScompact_13km_ics_HRRR_lbcs_RAP_suite_HRRR_20240104  COMPLETE              29.86
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta_2  COMPLETE              18.61
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_HRRR_suite_HRRR_2024010411522  COMPLETE              41.38
nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_timeoffset_suite_  COMPLETE              24.19
pregen_grid_orog_sfc_climo_20240104115227                          COMPLETE              22.15
specify_template_filenames_20240104115229                          COMPLETE              22.77
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             275.07
```

**Comment by MichaelLueken** [comment]
@mkavulich -

The `MET_verification_only_vx` WE2E test on Hercules failed in `run_MET_PcpCombine_obs_APCP01h` with the following message:
```
/work/noaa/epic/role-epic/jenkins/workspace/fs-srweather-app_pipeline_PR-997/hercules/scripts/exregional_run_met_pcpcombine.sh: line 140: 10#: invalid integer constant (error token is "10#")
End exregional_run_met_pcpcombine.sh at Thu Jan  4 17:28:02 UTC 2024 with error code 1 (time elapsed: 00:00:03)
```

**Comment by mkavulich** [comment]
@MichaelLueken Thanks for letting me know, can you point me to the location of Jenkins tests on Hercules? I'm not sure where to find that information, is it in the Wiki somewhere?

**Comment by MichaelLueken** [comment]
@mkavulich - The Jenkins workspace for your test on Hercules is:

`/work/noaa/epic/role-epic/jenkins/workspace/fs-srweather-app_pipeline_PR-997/hercules/expt_dirs/MET_verification_only_vx`

At the bottom of the [Contributor's Guide](https://github.com/ufs-community/ufs-srweather-app/wiki/Contributor's-Guide), there are paths to the Jenkins workspaces on the various machines.

**Comment by mkavulich** [comment]
@MichaelLueken For the record I am getting "Permission denied" errors when I attempt to view the contents of `/work/noaa/epic/role-epic/jenkins/workspace/fs-srweather-app_pipeline_PR-997/hercules/expt_dirs/MET_verification_only_vx`, can that be opened up in the future? I was able to replicate the error on my own so it's not hugely urgent

**Comment by MichaelLueken** [comment]
> @MichaelLueken For the record I am getting "Permission denied" errors when I attempt to view the contents of `/work/noaa/epic/role-epic/jenkins/workspace/fs-srweather-app_pipeline_PR-997/hercules/expt_dirs/MET_verification_only_vx`, can that be opened up in the future? I was able to replicate the error on my own so it's not hugely urgent

@mkavulich - Thanks for bringing this to my attention.  I have opened an issue with the EPIC Platform team to see if they can make the Jenkins workspace on Hercules readable for developers outside of the epic account.

**Comment by mkavulich** [comment]
@MichaelLueken This PR is ready for a re-test on Hercules. 

For anyone curious about the details: This was a very strange confluence of circumstances revealing an old bug where some logic in exregional_run_met_pcpcombine.sh that was only used for forecast jobs (not for observations) needed to be moved inside an if-block to avoid utilizing undefined variables (specifically, the variable `ENSMEM_INDX` doesn't make sense for observation tasks since they are independent of the number of ensemble members). This didn't cause an error until I introduced my Octal bug fix...and even then, for some reason this is *only* failing on Hercules: tests on Derecho and Hera still pass. So something specific about the bash version (or other environment) on Hercules is revealing more problems than other machines.

**Comment by MichaelLueken** [comment]
@mkavulich - The WE2E tests successfully passed on Hercules:
```
---------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used
----------------------------------------------------------------------------------------------------
custom_GFDLgrid__GFDLgrid_USE_NUM_CELLS_IN_FILENAMES_eq_FALSE_202  COMPLETE               8.54
grid_CONUS_25km_GFDLgrid_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_202  COMPLETE              10.44
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta_202  COMPLETE              30.58
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v17_p8_plot  COMPLETE              18.92
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR_2024011108  COMPLETE              26.41
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_RAP_20240111083  COMPLETE              54.92
grid_RRFS_CONUScompact_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_  COMPLETE              14.94
grid_RRFS_NA_13km_ics_FV3GFS_lbcs_FV3GFS_suite_RAP_20240111083520  COMPLETE              67.05
grid_SUBCONUS_Ind_3km_ics_NAM_lbcs_NAM_suite_GFS_v16_202401110835  COMPLETE              28.96
MET_verification_only_vx_20240111083521                            COMPLETE               0.22
specify_EXTRN_MDL_SYSBASEDIR_ICS_LBCS_20240111083523               COMPLETE               8.57
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             269.55
```
Merging this work now.

**Pull request merged**